File: week8/ParAbs.cpp
#include <iostream>
using namespace std;
// Abstract class
class Shape {
    protected:
    float dimension;
    public:
    void getDimension() 
    {
        cin >> dimension;
    }
    void display()
    {
        cout<<"I'm partial"<<endl;
    }
// pure virtual Function
    virtual float calculateArea() = 0;
};
class Square : public Shape {
    public:
    float calculateArea() 
    {
        return dimension * dimension;
    }
};
class Circle : public Shape {
    public:
    float calculateArea() 
    {
        return 3.14 * dimension * dimension;
    }
};
int main() {
    Square s;
    Circle c;
    s.display();
    cout << "Enter the length of the square : ";
    s.getDimension();
    c.display();
    cout << "Area of the square is : " << s.calculateArea() << endl;
    cout << "Enter the radius of circle : ";
    c.getDimension();
    cout << "Area of the circle is : " << c.calculateArea() << endl;
}

File: week8/ParAbs.java
abstract class Shape
{
    abstract float area();
    void display()
    {
        System.out.println("This is a shape");
    }
}
class Rectangle extends Shape
{
    float x,y;
    Rectangle(float base,float height)
    {
        x = base;
        y = height;
    }
        public float area()
        {
            return x*y;
        }
}
class Triangle extends Shape
{
    float x,y;
    Triangle(float base,float height)
    {
        x = base;
        y = height;
    }
    public float area()
    {
        return (float)0.50*x*y;
    }
}
public class PartialAbstraction
{
    public static void main(String[] args)
    {
        Rectangle obj1 = new Rectangle(5,10);
        Triangle obj2 = new Triangle(46,30);
        System.out.println("Area of the Rectangle is : "+obj1.area());
        System.out.println("Area of the Triangle is : "+obj2.area());
    }
}
